### PR TITLE
Remove double sort by channel from load_imgs_by_tree

### LINKED ISSUE #262: load_imgs_from_tree always sorts channels
https://github.com/angelolab/ark-analysis/blob/eaf429e6efa25e29babf463c89d6a3de52410a3f/ark/utils/load_utils.py#L247

The current implementation now always sorts fovs and channels. These means it's not possible to specify the order of channels in the xarray by passing in a list, since they'll get resorted anyway. I think before we had it set up s.t. sorting only happened if None was provided. Is there a reason that we need to sort in all cases?

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Addresses and closes #262.

**How did you implement your changes**

We need to remove the extra `.sorted('channels')` at the end of `load_imgs_by_tree`. This means the testing function will also need to be slightly altered.

### CODE REVIEW

File: ark/utils/load_utils.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def load_imgs_from_tree(data_dir, img_sub_folder=None, fovs=None, channels=None,
                           dims=["fovs", "rows", "cols", "channels"])
 
     # sort by fovs and channels for deterministic result
-    img_xr = img_xr.sortby('fovs').sortby('channels')
+    img_xr = img_xr.sortby('fovs')
```
Comment (ngreenwald) [inline ark/utils/load_utils.py:175]: Same thing for FOVs: I think we should sort once we list all of the FOVs in the folder if None is provided. If a custom FOV list is provided, then we shouldn't sort. If order matters for some reason, I think it makes sense to keep it in the order that the items were provided in the function call.

File: ark/utils/load_utils_test.py [hunk 2]
```diff
@@ -96,15 +96,22 @@ def test_load_imgs_from_tree():
             load_utils.load_imgs_from_tree(temp_dir, img_sub_folder="TIFs", dtype="int16",
                                            channels=some_chans)
 
-        assert loaded_xr.equals(data_xr[:, :, :, :2], )
+        # need to reindex columns first before asserting equality
+        # cannot use xarray.equals for this because reindexing string coords changes their dtype
+        assert np.all(loaded_xr.values == data_xr[:, :, :, :2].loc[
```
Comment (ngreenwald) [inline ark/utils/load_utils_test.py:101]: Why is this change necessary? We were sorting before and we're sorting now, so what's the difference?

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Nice start, see comments below

File: ark/utils/load_utils_test.py (hunk 2 above)
Comment (alex-l-kong) [reply ark/utils/load_utils_test.py:126]: So the problem seems to happen when `channels` is unordered (ex. `['chan1', 'chan0']`). This can either happen because the argument to `channels` itself is unordered or becomes unordered when not all of the channels have a proper extension (in which `iou.list_files` is run, and that potentially can return an unordered list as well).

In this case, the `channel` coords in `loaded_xr` will likely be ordered differently than it would be in the test data (`data_xr`) created by `create_paired_xarray_fovs`. If we want to test that the corresponding values are the same, we need to reindex one of these arrays prior to testing equality because `.equals` in `xarray` will complain that the `channel` coords do not match between the two xarrays.

File: ark/utils/load_utils.py (hunk 1 above)
Comment (ackagel) [reply ark/utils/load_utils.py:182]: I think @ngreenwald was suggesting to remove this line. fovs and channels are both already sorted if they are `None`, and this line always sorts by fovs, even with user specified order.

**Review by ackagel — COMMENTED** [review]
Not much to add.  Any reason those big binary files got added in?

`git add *` is a gift and a curse

File: ark/utils/load_utils.py (hunk 1 above)
Comment (alex-l-kong) [reply ark/utils/load_utils.py:182]: Yeah, forgot to do that. Will do for the next commit.

**Comment by alex-l-kong** [comment]
> Not much to add. Any reason those big binary files got added in?
> 
> `git add *` is a gift and a curse

Screwed up using Sublime Merge. Will remove for the next commit.

**Comment by alex-l-kong** [comment]
Oh whoops, sorry premature review request here. @ngreenwald perhaps we could use this to discuss the comment I left above about why we need to reindex `data_xr`.

File: ark/utils/load_utils_test.py (hunk 2 above)
Comment (ngreenwald) [reply ark/utils/load_utils_test.py]: Okay cool. In that case, maybe it would be worthwhile to add a few lines around 134 so that the channel names, with extensions, are in the same order as the supplied channels. I think it would be good for the channels, and fovs, to appear in the xarray in the order the user specified them.

**Comment by alex-l-kong** [comment]
Ok, this PR is finally ready for a real review!

File: ark/utils/load_utils_test.py [hunk 3]
```diff
@@ -131,6 +131,8 @@ def test_load_imgs_from_tree():
 
         assert loaded_xr.equals(data_xr)
 
+        # assert loaded_xr.equals(data_xr.loc[:, :, :, loaded_xr.coords['channels'].values])
```
Comment (ngreenwald) [inline ark/utils/load_utils_test.py:134]: What's going on with this guy

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Awesome. Can you doublecheck if any of the other data loaders do the same thing?

**Comment by alex-l-kong** [comment]
> Awesome. Can you doublecheck if any of the other data loaders do the same thing?

Yeah for sure. Looks like both `load_imgs_from_mibitiff` and `load_imgs_from_dir` also use an extra sort at the end. Time to nuke that.

**Review by ngreenwald — APPROVED** [review]
